pallets/maintenance-mode/src/types.rs

```diff
@@ -94,20 +94,11 @@ where
 	}
 
 	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::DispatchError> {
-		if Pallet::<T>::maintenance_mode() {
-			T::MaintenanceExecutiveHooks::pre_upgrade()
-		} else {
-			T::NormalExecutiveHooks::pre_upgrade()
-		}
-	}
-
-	#[cfg(feature = "try-runtime")]
-	fn post_upgrade(state: Vec<u8>) -> Result<(), sp_runtime::DispatchError> {
+	fn try_on_runtime_upgrade(checks: bool) -> Result<Weight, TryRuntimeError> {
```

Comment:
Is this not part of master?

Reply:
No, #10 was merged directly to the `moonbeam-polkadot-v1.1.0` branch. Maybe it's better to create a separate PR to cherrypick that into master and leave this PR only for the precompiles. What do you think?